dev/tests/integration/testsuite/Magento/MessageQueue/Model/ConsumerTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MessageQueue\Model;
+
+use Magento\Framework\MessageQueue\Consumer;
+use Magento\Framework\MessageQueue\ConsumerFactory;
+use Magento\Framework\MessageQueue\EnvelopeFactory;
+use Magento\Framework\MessageQueue\QueueInterface;
+use Magento\MysqlMq\Model\QueueManagement;
+use Magento\MysqlMq\Model\ResourceModel\Queue;
+use Magento\TestFramework\ObjectManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Tests the different cases of consumers running by Consumer processor
+ */
+class ConsumerTest extends TestCase
+{
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @var Consumer
+     */
+    private $model;
+
+    /**
+     * @var Queue
+     */
+    private $queueResource;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+        $this->objectManager = ObjectManager::getInstance();
+        /** @var ConsumerFactory $factory */
+        $factory = $this->objectManager->get(ConsumerFactory::class);
+        $this->model = $factory->get('demoConsumerWithConnectionException');
+        $this->queueResource = $this->objectManager->get(Queue::class);
+    }
+
+    /**
+     * Tests running of specific consumer check status if after retry is not in success status but left in progress
```

Comment:
This description is confusing. Please add more clear one.

Reply:
updated